src/stan/lang/grammars/semantic_actions_def.cpp

```diff
@@ -805,31 +795,26 @@ namespace stan {
     validate_void_return_allowed_f;
 
     void identifier_to_var::operator()(const std::string& name,
-                                       const var_origin& origin,
+                                       const scope& var_scope,
                                        variable& v,  bool& pass,
                                        const variable_map& vm,
                                        std::ostream& error_msgs) const {
       // validate existence
       if (!vm.exists(name)) {
-        pass = false;
+        pass = false;   // MM: no error message?  see validate_assignment below
         return;
       }
-      // validate origin
-      var_origin lhs_origin = vm.get_origin(name);
-      origin_block lhs_block = lhs_origin.program_block_;
-      if (lhs_block != local_origin
-          && lhs_block != origin.program_block_) {
+
+      // validate scope matches declaration scope or is nested in
+      scope lhs_origin = vm.get_scope(name);
+      if (lhs_origin.program_block() != var_scope.program_block()
+          && lhs_origin.program_block() != local_origin) {  // MM is_local()
```

Comment:
I dind't understand the top comment or the line comment in the code.  Don't leave your initials tracked through the code anywhere other than in TODO items.